pkg/services/validator/vmware.go

```diff
@@ -518,59 +408,93 @@ func configureEntityPrivilegeRules(ctx context.Context, c *components.VspherePlu
+		c.Validator.PrivilegeValidationRules[idx] = *r
 	}
 
 	return nil
 }
 
-func readEntityPrivileges(ctx context.Context, c *components.VspherePluginConfig, r *v1alpha1.EntityPrivilegeValidationRule, driver vsphere.Driver, isAdmin, reconfigureEntity bool) error {
+func readEntityPrivileges(ctx context.Context, c *components.VspherePluginConfig, r *v1alpha1.PrivilegeValidationRule, driver vsphere.Driver) error {
 	var err error
 
-	if isAdmin {
-		r.Username, err = prompts.ReadTextRegex("vSphere username to validate entity privileges for", r.Username, "Invalid vSphere username", cfg.VSphereUsernameRegex)
-		if err != nil {
-			return err
-		}
-	} else {
-		log.InfoCLI(`Privilege validation rule will be applied for username %s`, c.Validator.Auth.CloudAccount.Username)
-		r.Username = c.Validator.Auth.CloudAccount.Username
+	log.InfoCLI(`Privilege validation rule will be applied for username %s`, c.Validator.Auth.Account.Username)
+
+	r.EntityType, err = prompts.Select("Entity Type", entity.Labels)
+	if err != nil {
+		return err
 	}
 
-	if reconfigureEntity {
-		r.EntityType, r.EntityName, r.ClusterName, err = getEntityInfo(ctx, "", "Entity Type", c.Validator.Datacenter, cfg.ValidatorPluginVsphereEntities, driver)
-		if err != nil {
-			return err
-		}
-		r.Privileges, err = readPrivileges(r.Privileges)
-		if err != nil {
-			return err
-		}
+	r.EntityName, r.ClusterName, err = getEntityAndClusterInfo(ctx, r.EntityType, driver, c.Validator.Datacenter)
+	if err != nil {
+		return err
+	}
+
+	r.Privileges, err = readPrivileges(r.Privileges)
+	if err != nil {
+		return err
+	}
+
+	log.InfoCLI(`
+	Propagation validation identifies the permission(s) that grant privileges to the user
+	with respect to the rule's entity and verifies the value of their propagation flag.
+	`)
+	r.Propagation.Enabled, err = prompts.ReadBool("Enable propagation validation", false)
+	if err != nil {
+		return err
+	}
+	if !r.Propagation.Enabled {
+		return nil
+	}
+
+	log.InfoCLI(`
+	Provide a list of group principals to consider during validation. Group principals
+	should be of the format, DOMAIN\group-name. It is recommended to provide a group principal for
+	each group that the vCenter user is a member of. The user's own principal is included by default.
+	`)
+	r.Propagation.GroupPrincipals, err = prompts.ReadTextSlice(
+		"Group Principals", strings.Join(r.Propagation.GroupPrincipals, "\n"), "", "", true,
+	)
+	if err != nil {
+		return err
+	}
+
+	log.InfoCLI(`
+	Is the permission that assigns privileges to the user for the selected entity expected to
+	have propagation enabled? If multiple permissions for group principals associated with the
+	user exist on the entity in question, propagation will be consider enabled if any of them
+	has propagation enabled. If a permission exists for the entity whose principal is the user,
+	the propagation value of that permission takes precedence.
+	`)
+	r.Propagation.Propagated, err = prompts.ReadBool("Propagated", false)
```

Comment:
should this only be prompted for if `propagation validation` is enabled above? ie if propogation validation is _not_ enabled, then users probably wouldn't care if an entity is expected to have propagation enabled.

Reply:
There's an early return for that.